### PR TITLE
proof of concept - attrs TrainerParameters

### PR DESCRIPTION
**Pull request by chriselion**
Any thoughts on converting to something like this? I think it would let us simplify the yaml config files (remove anything that's set to the default) and clean up some of the dictionary-lookup code like https://github.com/Unity-Technologies/ml-agents/blob/e6dff56adfb99cca5c8854f66603e1a964fcdab4/ml-agents/mlagents/trainers/ppo/policy.py#L34-L48

### CODE REVIEW

File: ml-agents/mlagents/trainers/config.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+@attr.s(auto_attribs=True)
+class TrainerParameters:
+    trainer: str = "ppo"  # TODO enum
+    batch_size: int = 1024
+    beta: float = 5.0e-3
+    buffer_size: int = 10240
+    epsilon: float = 0.2
+    gamma: float = .99  # TODO is this the current default?
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/config.py]: We can remove this as it's defined in the reward signals

Comment (chriselion) [reply ml-agents/mlagents/trainers/config.py]: Ah, I think a few configs are still passing it in :)

**Comment by ervteng** [comment]
This looks pretty good. One consideration is that each trainer type (e.g. ppo, sac, online_bc, etc.) has different parameters and different defaults. So maybe we need a different TrainerParameters object for each type rather than one for all.

File: ml-agents/mlagents/trainers/config.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/config.py]: Uh oh, that's a bug - I removed them and pushed it to #2296

**Pull request closed without merging**